ext/phar/tar.c

```diff
@@ -181,9 +181,15 @@ static int phar_tar_process_metadata(phar_entry_info *entry, php_stream *fp) /*
 	}
 
 	if (entry->filename_len == sizeof(".phar/.metadata.bin")-1 && !memcmp(entry->filename, ".phar/.metadata.bin", sizeof(".phar/.metadata.bin")-1)) {
+		if (Z_TYPE(entry->phar->metadata) != IS_UNDEF) {
```

Comment:
We should do this check earlier, before we parse, I think.

Reply:
Wouldn't that require to duplicate the check whether the metadata belong to an entry or the whole phar (or to remember the result of that check)? Maybe it's more straight forward to free the parsed metadata before returning?